BraintreeCore/src/main/java/com/braintreepayments/api/core/BraintreeHttpRequest.kt

```diff
@@ -0,0 +1,17 @@
+package com.braintreepayments.api.core
+
+import androidx.annotation.RestrictTo
+import com.braintreepayments.api.sharedutils.HttpClient
+import com.braintreepayments.api.sharedutils.HttpClient.RetryStrategy
+
+/**
+ * @suppress
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+data class BraintreeHttpRequest(
+    val method: String,
+    val path: String,
+    val data: String? = null,
+    val additionalHeaders: Map<String, String> = emptyMap(),
+    @RetryStrategy val retryStrategy: Int = HttpClient.NO_RETRY
```

Comment:
Not sure if it's in scope for this PR - should we update this to an enum type?

Reply:
Yeah actually I was thinking of removing this enum all together in a follow up PR. The `ConfigurationLoader` is the only class that needs retry logic, so the end-goal is to fully encapsulate retries within the config loader.